add docs for MADS TLS (#1055)
kumahq/kuma-website: docs/docs/dev/security/certificates.md

See [Data plane proxy authentication](./dp-auth.md) and [Zone Ingress authentication](./zone-ingress-auth.md).

> ## Prometheus to control plane communication

Reviewer: Just a note, for 2.0, this doc was moved to `src/features/zero-trust/certificates.md`

> It is possible to enabled TLS on the Monitoring Assignment Discovery Service.

Reviewer 2: as a first-time reader unfamiliar with the product, I don't know what it is. A quick search for "Monitoring Assignment Discovery Service" does not help :(

> It is possible to enabled TLS on the Monitoring Assignment Discovery Service.

Reviewer: ```suggestion
You can enable TLS on the Monitoring Assignment Discovery Service.
```

> By default it will use the certificate used in `tls-general` but TLS will be disabled.

Reviewer: ```suggestion
By default, it uses the certificate in `tls-general`, but TLS will be disabled.
```

> You can enable it by using either the environment variable `KUMA_MONITORING_ASSIGNMENT_SERVER_TLS_ENABLED=true` or the yaml config `monitoringAssignmentServer.tlsEnabled=true`.

Reviewer 2: My strategy for referencing config is to use always env var to not repeat myself and aggressively link to `manage->configuration` so people understand how to set this up and what is an equivalent in YAML.

> You can enable it by using either the environment variable `KUMA_MONITORING_ASSIGNMENT_SERVER_TLS_ENABLED=true` or the yaml config `monitoringAssignmentServer.tlsEnabled=true`.

Reviewer: ```suggestion
You can enable it by using either the `KUMA_MONITORING_ASSIGNMENT_SERVER_TLS_ENABLED=true` environment variable or the `monitoringAssignmentServer.tlsEnabled=true` yaml config.
```

:::: tabs :options="{ useUrlFragment: false }"

> ::: tab "Kubernetes (kumactl)"

Reviewer: If this is for 2.0, I'm not sure if these tab options will work in Jekyll. Here's a doc on how we format tabs on a page in Jekyll: https://docs.konghq.com/contributing/markdown-rules/#tabs

Create a secret in the namespace where the control plane is installed:
```sh
kubectl create secret generic general-tls-certs -n <namespace> \
  --from-file=tls.crt=/tmp/tls.crt \
  --from-file=tls.key=/tmp/tls.key \
  --from-file=ca.crt=/tmp/ca.crt
```

> Point to this secret when installing Kuma:

Reviewer: ```suggestion
Point to this secret when installing {{site.mesh_product_name}}:
```
If this is for 2.0. We created this variable (not merged yet) to swap Kuma/Kong Mesh on shared docs.

```sh
kumactl install control-plane \
  --tls-general-secret=general-tls-certs \
  --tls-general-ca-bundle=$(cat /tmp/ca.crt | base64) \
  --env-var 'KUMA_MONITORING_ASSIGNMENT_SERVER_TLS_ENABLED=true'
```

:::

> ::: tab "Universal"

Reviewer: Same comment here about that this might not work for tabs in 2.0.

Configure the control plane with generated certificates:

```sh
KUMA_GENERAL_TLS_CERT_FILE=/tmp/tls.crt \
  KUMA_GENERAL_TLS_KEY_FILE=/tmp/tls.key \
  KUMA_MONITORING_ASSIGNMENT_SERVER_TLS_ENABLED=true \
  kuma-cp run

> ```

Reviewer 2: Should the instruction be about general cert or mads cert?

:::
::::

> You can then configure Kuma's prometheus SD with the right TLS configuration using the [Prometheus docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kuma_sd_config).

Reviewer: ```suggestion
Now you can configure Kuma's Prometheus SD with the correct TLS configuration using the [Prometheus docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kuma_sd_config).
```
I'm not sure what "SD" stands for here, but if it's an acronym I'd spell it out first like "Prometheus Service Discovery (SD)".
